cst438week2_assignment2/src/main/java/cst438_assignment2/domain/CityInfo.java

```diff
@@ -33,9 +29,24 @@ public CityInfo(City city, TempAndTime tempAndTime) {
 		this.time = convertTime(tempAndTime);
 	}
 	
+	public CityInfo(int id, String name, String countryCode, String countryName, String district, int population,
+			double temp, String time) {
+		super();
+		this.id = id;
+		this.name = name;
+		this.countryCode = countryCode;
+		this.countryName = countryName;
+		this.district = district;
+		this.population = population;
+		this.temp = temp;
+		this.time = time;
+	}
+
+
+
 	public String convertTime(TempAndTime t) {
 
-		Date date = new Date((long)(t.getTime()+t.getTimezone() )*1000);
+		Date date = new Date((long)(t.getTime() - t.getTimezone()  )*1000);
 		DateFormat format = new SimpleDateFormat("hh:mm a");
 		return format.format(date);
```

Comment:
The offset should be how much you add to the getTime from getTimezone. Unfortunately I still run into the same error as well